### PR TITLE
Allows admins to delete notes they made under all circumstances

### LINKED ISSUE #24282: Player notes compare key instead of ckey
Admins are supposed to be able to delete player notes that they have added, but sometimes the option might not appear because `key` is [not guaranteed to be the same](http://www.byond.com/docs/ref/#/mob/var/key).

This needs to be changed to use `ckey`.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Closes #24282  
[bugfix]

### CODE REVIEW

**Review by DamianX — CHANGES_REQUESTED** [review]
Notes still keep saving key instead of ckey, so now you're comparing the stored `key` vs `usr.ckey`

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -399,7 +399,7 @@ var/global/floorIsLava = 0
 				I.rank = "N/A"
 				update_file = 1
 			dat += "<font color=#008800>[I.content]</font> <i>by [I.author] ([I.rank])</i> on <i><font color=blue>[I.timestamp]</i></font> "
-			if(I.author == usr.key || check_rights(R_PERMISSIONS, show_msg = 0))
+			if(ckey(I.author) == usr.ckey || check_rights(R_PERMISSIONS, show_msg = 0)) // Older notes stored the key of the banning admin.
```
Comment (DamianX) [inline code/modules/admin/admin.dm:402]: > Older

new notes would also use `key`, it wasn't changed

**Comment by Killette2** [comment]
That's right goyim, you can now get your notes removed after a small donation to a charity I've made

File: code/modules/admin/admin.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/modules/admin/admin.dm:402]: I thought this wouldn't be a problem since now, only the "canonical" key would be compared. I will fix it.

Comment (DamianX) [reply code/modules/admin/admin.dm:402]: It's not a problem in the sense that this works, but the comment is misleading.

**Review by DamianX — APPROVED** [review]

**Pull request merged**